lib/shopify-cli/commands/connect.rb

```diff
@@ -3,23 +3,48 @@
 module ShopifyCli
   module Commands
     class Connect < ShopifyCli::Command
-      def call(*)
-        project_type = ask_project_type unless Project.has_current?
+      class << self
+        def call(args, command_name)
+          ProjectType.load_type(args[0]) unless args.empty?
+          super
+        end
 
-        if Project.has_current? && Project.current && Project.current.env
-          @ctx.puts @ctx.message('core.connect.already_connected_warning')
-          prod_warning = @ctx.message('core.connect.production_warning')
-          @ctx.puts prod_warning if [:rails, :node].include?(Project.current_project_type)
+        def default_connect(project_type)
+          org = ShopifyCli::Tasks::EnsureEnv.call(@ctx, regenerate: true)
+          write_cli_yml(project_type, org['id']) unless Project.has_current?
+          api_key = Project.current(force_reload: true).env['api_key']
+          [org, api_key]
+        end
+
+        def write_cli_yml(project_type, org_id)
+          ShopifyCli::Project.write(
+            @ctx,
+            project_type: project_type,
+            organization_id: org_id,
+          )
+          @ctx.done(@ctx.message('core.connect.cli_yml_saved'))
         end
 
-        org = ShopifyCli::Tasks::EnsureEnv.call(@ctx, regenerate: true)
-        write_cli_yml(project_type, org['id']) unless Project.has_current?
-        api_key = Project.current(force_reload: true).env['api_key']
-        @ctx.puts(@ctx.message('core.connect.connected', get_app(org['apps'], api_key).first["title"]))
+        def help
+          ShopifyCli::Context.message('core.connect.help', ShopifyCli::TOOL_NAME)
+        end
       end
 
-      def get_app(apps, api_key)
-        apps.select { |app| app["apiKey"] == api_key }
+      def call(args, command_name)
+        if Project.has_current? && Project.current && Project.current.env
```

Comment:
Can this be made simpler?

```suggestion
        if Project.current&.env
```